app/rest/response_json.h

```diff
@@ -74,12 +74,26 @@ class ResponseJson : public Response {
     // Parse and verify JSON string in body. FlatBuffer parser does not support
     // online parsing. So we only parse the body when we get everything.
     bool parse_status = parser_->Parse(GetBody());
-    FIREBASE_ASSERT_RETURN_VOID(parse_status);
+    FIREBASE_ASSERT_MESSAGE(parse_status,
```

Comment:
With FIREBASE_ASSERT_MESSAGE, it will assert in C++ and throw an exception in Unity. Is that the behavior we want? Maybe this should just be a LogError. Same below.

Reply:
I'm not sure if LogError() would be more appropriate. Assert in C++ and throw in Unity sounds reasonable though.

Note that both FIREBASE_ASSERT_RETURN_VOID and FIREBASE_ASSERT_MESSAGE use firebase::LogAssert() under the hood to log the assertion and crash/throw/ignore the failed assertion. So the change made in this PR does not change that behavior. If that should be changed then I think that should be a separate PR since I don't know the right answer and only learned enough about this Auth code to fix the bug at hand.

Reply:
Well, what I'm saying is, asserting at all (in C++) on bad data coming in (outside the app's control) seems like a bad idea, no?

Reply:
Totally agree. It's an inappropriate use of assertions since the data is out of the app's control. I guess the rationale is that using assertions makes it VERY visible what's going wrong. Otherwise, if we change it to just LogError(), it would be more difficult to correlate the ultimate failure with the root cause of a parsing failure. But I guess the fact that checking for a zero-length body earlier in the method and not even logging anything gives merit to the argument that we shouldn't crash. Ok, I'll change it to LogError().

Reply:
Thanks - yeah this fulfills the software [robustness principle](https://en.wikipedia.org/wiki/Robustness_principle): "Be conservative in what you do, be liberal in what you accept from others."